apps/web/src/utils/datastores/kv/index.ts

```diff
@@ -0,0 +1,179 @@
+import Redis, { type Redis as RedisType } from 'ioredis';
+import { isDevelopment } from 'apps/web/src/constants';
+import { logger } from '../../logger';
+
+type KvConstructorParam =
+  | { url: string; tls?: boolean }
+  | { host: string; port: number; username?: string; password?: string; tls?: boolean };
+
+/**
+ * Provides a limited, type-safe interface to Redis operations.
+ * Intentionally restricts access to dangerous commands and raw client operations.
+ */
+export class KVManager {
```

Comment:
suggestion(non-blocking): Since we're only instantiating once, you could skip exporting the class and just export the singleton

Reply:
good call. i originally had this exported so that i could create CBHQ clients as well, but i'm going to make that client a singleton also, so i'll remove this export